### PR TITLE
fix(ci): use dedicated rosetta config dir

### LINKED ISSUE #4472: Fix rosetta CI
The rosetta validation CI job is failing due to not being able to run genesis. https://github.com/iotaledger/iota/actions/runs/12281205786/job/34309129803?pr=4446

```
run IOTA genesis
Cannot run genesis with non-empty Iota config directory /home/runner/.iota/iota_config, please use the --force/-f option to remove the existing configuration

Stack backtrace:
   0: anyhow::error::<impl anyhow::Error>::msg
   1: iota::iota_commands::genesis::{{closure}}
   2: iota::iota_commands::IotaCommand::execute::{{closure}}
   3: tokio::runtime::park::CachedParkThread::block_on
   4: tokio::runtime::context::runtime::enter_runtime
   5: tokio::runtime::runtime::Runtime::block_on_inner
   6: iota::main
   7: std::sys::backtrace::__rust_begin_short_backtrace
   8: std::rt::lang_start::{{closure}}
   9: std::rt::lang_start_internal
  10: main
  11: <unknown>
  12: __libc_start_main
  13: _start
generate rosetta configuration
install rosetta-cli
coinbase/rosetta-cli info checking GitHub for latest tag
coinbase/rosetta-cli info found version: 0.10.4 for v0.10.4/linux/amd64
```

### PR DESCRIPTION
**Pull request by kodemartin**
# Description of change

Fixes rosetta setup in the CI by forcing a dedicated config dir for `iota genesis` in `.github/scripts/rosetta/setup.sh`.

## Links to any relevant issues

Fixes #4472 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

### CODE REVIEW

**Review by alexsporn — APPROVED** [review]

**Comment by DaughterOfMars** [comment]
I pushed a change to enable running this test when only rosetta files are changed. Let me know if that's okay. If not, I can move it to a new PR.

**Review by thibault-martinez — APPROVED** [review]
LGTM

File: .github/workflows/_rosetta.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  # Allow more retries for network requests in cargo (downloading crates) and
+  # rustup (installing toolchains). This should help to reduce flaky CI failures
+  # from transient network timeouts or other issues.
+  CARGO_NET_RETRY: 10
+  RUSTUP_MAX_RETRIES: 10
+  # RUSTFLAGS: -D warnings
+  RUSTDOCFLAGS: -D warnings
+
+jobs:
+  test:
```
Comment (thibault-martinez) [inline .github/workflows/_rosetta.yml]: I'd still call it `validation` like before though, the process is a bit more intense than just executing a test. Just resolve if you prefer `test`.

Comment (DaughterOfMars) [reply .github/workflows/_rosetta.yml]: done

**Pull request merged**